src/libasr/codegen/asr_to_python.cpp

```diff
@@ -0,0 +1,634 @@
+        }
+
+        // Main program
+        for (auto &item : x.m_symtab->get_scope()) {
+            if (is_a<ASR::Program_t>(*item.second)) {
+                visit_symbol(*item.second);
+                r += s;
+            }
+        }
+        s = r;
+    }
+
+    void visit_Module(const ASR::Module_t &x) {
+        std::string r;
+
+        for (auto &item : x.m_symtab->get_scope()) {
+            if (is_a<ASR::Function_t>(*item.second)) {
+                visit_symbol(*item.second);
+                r += s;
+            }
+        }
+        s = r;
+    }
+
+    void visit_Function(const ASR::Function_t &x) {
+        // Generate code for the lpython function
+        std::string r;
+        r = "def";
+        r += " ";
+        r.append(x.m_name);
+        r += "(";
+        for (size_t i = 0; i < x.n_args; i++) {
+            visit_expr(*x.m_args[i]);
+            r += s;
+            if (i < x.n_args - 1) {
+                r += ", ";
+            }
+        }
+        r += "):";
+        r += "\n";
+
+        inc_indent();
+        for (auto &item : x.m_symtab->get_scope()) {
+            if (is_a<ASR::Variable_t>(*item.second)) {
+                visit_symbol(*item.second);
+                r += s;
+            }
+        }
+        dec_indent();
+
+        visit_body(x, r, true);
+
+        s = r;
+    }
+
+    void visit_Program(const ASR::Program_t &x) {
+        std::string r;
+
+        for (auto &item : x.m_symtab->get_scope()) {
+            if (is_a<ASR::Function_t>(*item.second)) {
+                visit_symbol(*item.second);
+                r += s;
+            }
+        }
+
+        r += "\n";
+        s = r;
+    }
+
+    void visit_Variable(const ASR::Variable_t &x) {
+        std::string r = indent;
+        r += x.m_name;
+        r += ": ";
+        r += get_type(x.m_type);
+        r += "\n";
+        s = r;
+    }
+
+    void visit_Print(const ASR::Print_t &x) {
+        std::string r = indent;
+        r += "print(";
+        for (size_t i = 0; i < x.n_values; i++) {
+            visit_expr(*x.m_values[i]);
+            r += s;
+            if (i < x.n_values-1)
+                r += ", ";
+        }
+        r += ")";
+        r += "\n";
+        s = r;
+    }
+
+    void visit_Assignment(const ASR::Assignment_t &x) {
+        std::string r = indent;
+        visit_expr(*x.m_target);
+        r += s;
+        r += " = ";
+        visit_expr(*x.m_value);
+        r += s;
+        r += "\n";
+        s = r;
+    }
+
+    void visit_SubroutineCall(const ASR::SubroutineCall_t &x) {
+        std::string r = indent;
+        r += ASRUtils::symbol_name(x.m_name);
+        r += "(";
+        for (size_t i = 0; i < x.n_args; i++) {
+            visit_expr(*x.m_args[i].m_value);
+            r += s;
+            if (i < x.n_args - 1)
+                r += ", ";
+        }
+        r += ")\n";
+        s = r;
+    }
+
+    void visit_FunctionCall(const ASR::FunctionCall_t &x) {
+        std::string r = indent;
```

Comment:
```suggestion
        std::string r = "";
```
`FunctionCall` is an expr, so no need for indent.